exonum/src/node/consensus.rs

```diff
@@ -556,9 +550,16 @@ impl NodeHandler {
         let full_proposes = self.state.check_incomplete_proposes(hash);
         // Go to handle full propose if we get last transaction
         for (hash, round) in full_proposes {
-            self.remove_request(&RequestData::Transactions(hash));
+            self.remove_request(&RequestData::TransactionsForPropose(hash));
             self.handle_full_propose(hash, round);
         }
+
+        let full_blocks = self.state.check_incomplete_blocks(hash);
+        // Go to handle full block if we get last transaction
+        for (hash, block) in full_blocks {
```

Comment:
I'm *so* confused by this code. Can there be multiple blocks committed here? If yes, how does this work? If no, why use a cycle?